From 1fc00dd2dc6f4dfabf30ebe7c531841f8ecda5d7 Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@codasip.com>
Date: Thu, 16 May 2024 17:48:14 +0100
Subject: [PATCH 15/15] riscv: configs: hobgoblin: Convert isa detection to new
 properties

Convert the Hobgoblin devicetree to use the new properties
"riscv,isa-base" & "riscv,isa-extensions".
For compatibility with other projects, "riscv,isa" remains.

Signed-off-by: Stuart Menefy <stuart.menefy@codasip.com>
---
 arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts      | 4 ++++
 arch/riscv/configs/codasip-a70x-hobgoblin_defconfig | 1 +
 2 files changed, 5 insertions(+)

diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
index 767618b3c743..7e6e3ea77be0 100644
--- a/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
@@ -43,6 +43,10 @@ cpu0: cpu@0 {
 			device_type = "cpu";
 			reg = <0>;
 			riscv,isa = "rv64imafdc_svpbmt_zicbom_zicboz_zicbop";
+			riscv,isa-base = "rv64i";
+			riscv,isa-extensions = "i", "m", "a", "f", "d", "c",
+					       "svpbmt", "zicbom", "zicboz",
+					       "zicbop";
 			riscv,cbom-block-size = <64>;
 			riscv,cboz-block-size = <64>;
 			mmu-type = "riscv,sv39";
diff --git a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
index 9827c6d41d4e..5266d28ed64e 100644
--- a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
+++ b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
@@ -21,6 +21,7 @@ CONFIG_RISCV_EFFICIENT_UNALIGNED_ACCESS=y
 CONFIG_HZ_100=y
 # CONFIG_COMPAT is not set
 # CONFIG_EFI is not set
+# CONFIG_RISCV_ISA_FALLBACK is not set
 # CONFIG_SUSPEND is not set
 CONFIG_CPU_IDLE=y
 CONFIG_RISCV_SBI_CPUIDLE=y
-- 
2.39.3

